models/Document/Dao.php

```diff
@@ -377,11 +377,21 @@ public function getChildAmount($user = null)
     /**
      * Checks if the document has siblings
      *
+     * @param bool $unpublished
+     *
      * @return bool
      */
-    public function hasSiblings()
+    public function hasSiblings($unpublished = false)
     {
-        $c = $this->db->fetchOne('SELECT id FROM documents WHERE parentId = ? and id != ? LIMIT 1', [$this->model->getParentId(), $this->model->getId()]);
+        $sql = 'SELECT id FROM documents WHERE parentId = ? and id != ?';
+
+        if (Model\Document::doHideUnpublished() && !$unpublished) {
+            $sql .= ' AND published = 1';
```

Comment:
meaning that I can set the flag to whatever I want if the first check isn't satisfied ?

Reply:
We should probably use `null` as the default value for `$unpublished` then we can check whether it was set or if the default from `doHideUnpublished()` should be used.

Reply:
done!